airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/source.py

```diff
@@ -0,0 +1,362 @@
+        https://docs.microsoft.com/en-us/linkedin/shared/api-guide/concepts/pagination?context=linkedin/marketing/context
+        """
+        parsed_response = response.json()
+        if len(parsed_response.get("elements")) < self.records_limit:
+            return None
+        return {"start": parsed_response.get("paging").get("start") + self.records_limit}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = {"count": self.records_limit, "q": "search"}
+        if next_page_token:
+            params.update(**next_page_token)
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        We need to get out the nested complex data structures for further normalisation, so the transform_data method is applied.
+        """
+        yield from transform_data(response.json().get("elements"))
+
+
+class Accounts(LinkedinAdsStream):
+    """
+    Get Accounts data. More info about LinkedIn Ads / Accounts:
+    https://docs.microsoft.com/en-us/linkedin/marketing/integrations/ads/account-structure/create-and-manage-accounts?tabs=http
+    """
+
+    endpoint = "adAccountsV2"
+
+    def request_headers(self, stream_state: Mapping[str, Any], **kwargs) -> Mapping[str, Any]:
+        """
+        If account_ids are specified as user's input from configuration,
+        we must use MODIFIED header: {'X-RestLi-Protocol-Version': '2.0.0'}
+        """
+        return {"X-RestLi-Protocol-Version": "2.0.0"} if self.accounts else {}
+
+    def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        """
+        Override request_params() to have the ability to accept the specific account_ids from user's configuration.
+        If we have list of account_ids, we need to make sure that the request_params are encoded correctly,
+        We will get HTTP Error 500, if we use standard requests.urlencode methods to parse parameters,
+        so the urlencode(..., safe=":(),") is used instead, to keep the values as they are.
+        """
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        if self.accounts:
+            account_list = ",".join(map(str, self.accounts))
+            params["search"] = f"(id:(values:List({account_list})))"
+            return urlencode(params, safe=":(),")
+        return params
+
+
+class IncrementalLinkedinAdsStream(LinkedinAdsStream):
+
+    cursor_field = "lastModified"
+
+    @property
+    def primary_slice_key(self) -> str:
+        """
+        Define the main slice_key from `slice_key_value_map`. Always the first element.
+        EXAMPLE:
+            in : {"k1": "v1", "k2": "v2", ...}
+            out : "k1"
+        """
+        return list(self.slice_key_value_map.keys())[0]
+
+    @property
```

Comment:
Which one?

Reply:
This property is needed, because of it's usability, we can fetch the first key from `slice_key_value_map` if it's present, rather than hardcode the logic for each stream class inside of `request_params` to get the proper `key` from stream slice.